pkg/resources/istioingress/virtualservice.go

```diff
@@ -27,52 +27,71 @@ import (
 	kafkautils "github.com/banzaicloud/kafka-operator/pkg/util/kafka"
 )
 
-func (r *Reconciler) virtualService(log logr.Logger, externalListenerConfig v1beta1.ExternalListenerConfig) runtime.Object {
+func (r *Reconciler) virtualService(log logr.Logger, externalListenerConfig v1beta1.ExternalListenerConfig,
+	ingressConfig v1beta1.IngressConfig, ingressConfigName, defaultIngressConfigName string) runtime.Object {
+
+	var gatewayName, virtualSName string
+	if ingressConfigName == util.IngressConfigGlobalName {
+		gatewayName = fmt.Sprintf(gatewayNameTemplate, r.KafkaCluster.Name, externalListenerConfig.Name)
+		virtualSName = fmt.Sprintf(virtualServiceTemplate, r.KafkaCluster.Name, externalListenerConfig.Name)
+	} else {
+		gatewayName = fmt.Sprintf(gatewayNameTemplateWithScope, r.KafkaCluster.Name, externalListenerConfig.Name, ingressConfigName)
+		virtualSName = fmt.Sprintf(virtualServiceTemplateWithScope, r.KafkaCluster.Name, externalListenerConfig.Name, ingressConfigName)
+	}
+
 	vServiceSpec := v1alpha3.VirtualServiceSpec{
 		Hosts:    []string{"*"},
-		Gateways: []string{fmt.Sprintf(gatewayNameTemplate, r.KafkaCluster.Name, externalListenerConfig.Name)},
+		Gateways: []string{gatewayName},
 	}
 
-	if r.KafkaCluster.Spec.IstioIngressConfig.TLSOptions != nil &&
-		r.KafkaCluster.Spec.IstioIngressConfig.TLSOptions.Mode == v1alpha3.TLSModePassThrough {
-		vServiceSpec.TLS = generateTlsRoutes(r.KafkaCluster, externalListenerConfig, log)
+	if ingressConfig.IstioIngressConfig.TLSOptions != nil &&
+		ingressConfig.IstioIngressConfig.TLSOptions.Mode == v1alpha3.TLSModePassThrough {
+		vServiceSpec.TLS = generateTlsRoutes(r.KafkaCluster, externalListenerConfig, log, ingressConfigName, defaultIngressConfigName)
 
 	} else {
-		vServiceSpec.TCP = generateTcpRoutes(r.KafkaCluster, externalListenerConfig, log)
+		vServiceSpec.TCP = generateTcpRoutes(r.KafkaCluster, externalListenerConfig, log, ingressConfigName, defaultIngressConfigName)
 	}
 
 	return &v1alpha3.VirtualService{
 		ObjectMeta: templates.ObjectMetaWithAnnotations(
-			fmt.Sprintf(virtualServiceTemplate, r.KafkaCluster.Name, externalListenerConfig.Name),
-			labelsForIstioIngress(r.KafkaCluster.Name, externalListenerConfig.Name),
-			r.KafkaCluster.Spec.IstioIngressConfig.GetVirtualServiceAnnotations(),
+			virtualSName,
+			labelsForIstioIngress(r.KafkaCluster.Name, annotationName),
```

Comment:
Why `annotationName` is passed into `labelsForIstioIngress` instead of `externalListenerConfig.Name`?